src/lib.rs

```diff
@@ -654,4 +714,64 @@ mod tests {
         assert_eq!(sol[v2], 1.0);
         assert_eq!(sol.objective(), -1.0);
     }
+
+    #[test]
+    fn knapsak_solve() {
+        let mut problem = Problem::new(OptimizationDirection::Maximize);
+        let weights = [10, 60, 30, 40, 30, 20, 20, 2];
+        let values = [1, 10, 15, 40, 60, 90, 100, 15];
+        let capacity = 102;
+        let mut vars = vec![];
+        for i in 0..weights.len() {
+            let var = problem.add_binary_var(values[i] as f64);
+            vars.push(var);
+        }
+        let entries = vars
+            .iter()
+            .map(|v| (*v, weights[v.0] as f64))
+            .collect::<Vec<_>>();
+        problem.add_constraint(&entries, ComparisonOp::Le, capacity as f64);
+        let sol = problem.solve().unwrap();
+        let values = vars.iter().map(|v| sol[*v]).collect::<Vec<_>>();
+        assert_eq!(
+            values,
+            vec![0.0, 0.0, 0.999999999999999, 0.0, 1.0, 1.0, 1.0, 1.0]
```

Comment:
are you sure we want the assert with 0.999999999999999 ?

Reply:
This was more of a lazy way to just make the test pass. 
But regarding this, once a solution is found, should we round the number to it's integer value or leave it with rounding errors?

Rounding might cause accuracy issues if the solution is reused to add more constraints as we'd have to replace them in the solver state itself.

Or maybe a separate method used to get the variable values which inside clamps only integer variables?

Or nothing at all and we just document the precision issues?

Reply:
In this case all the variables are integers, I would just cast `values` to a Vec<u8> before the assert.

You could also add a `sol.integer_value(variable)` method.